zip/zlib.nim

```diff
@@ -222,36 +228,39 @@ proc compress*(sourceBuf: cstring; sourceLen: int; level=Z_DEFAULT_COMPRESSION;
   var status = deflateInit2(z, level.int32, Z_DEFLATED.int32,
                                windowBits.int32, Z_MEM_LEVEL.int32,
                                Z_DEFAULT_STRATEGY.int32)
-
-  if status != Z_OK:
-    # Out of memory.
-    return
+  case status
+  of Z_OK: discard
+  of Z_MEM_ERROR: raise newException(OutOfMemError, "")
+  of Z_STREAM_ERROR: raise newException(ZlibStreamError, "invalid zlib stream parameter!")
+  of Z_VERSION_ERROR: raise newException(ZlibStreamError, "zlib version mismatch!")
+  else: raise newException(ZlibStreamError, "Unkown error(" & $status & ") : " & $z.msg)
 
   let space = deflateBound(z, sourceLen)
-
   var compressed = newStringOfCap(space)
   z.next_in = sourceBuf
   z.avail_in = sourceLen.Uint
   z.next_out = addr(compressed[0])
   z.avail_out = space.Uint
 
-  defer: discard deflateEnd(z)
-
   status = deflate(z, Z_FINISH)
-
   if status != Z_STREAM_END:
-    # Incomplete stream.
-    return
+    discard deflateEnd(z) # cleanup allocated ressources
```

Comment:
you should also verify that ``deflateEnd`` succeeded by checking its return value, that's what I meant in my previous comment.